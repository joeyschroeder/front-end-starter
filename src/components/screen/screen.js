import React, { Component } from 'react';

import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './screen.style.scss';

export class Screen extends Component {
  static propTypes = {
    centerContent: PropTypes.bool,
    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
    className: PropTypes.string
  };

  static defaultProps = {
    centerContent: false,
    children: null,
    className: ''
  };

  getClassNames() {
    const { centerContent, className } = this.props;

    const result = classNames('animated', 'fadeIn', className, styles.root, { [styles.centerContent]: centerContent });

    return result;
  }

  render() {
    const { children } = this.props;
    const className = this.getClassNames();

    return <div className={className}>{children}</div>;
  }
}
