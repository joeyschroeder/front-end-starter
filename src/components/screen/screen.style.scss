@import "../../styles/config/variables";
@import "../../styles/mixins/transition";

.root {
  @include transition(background-color);

  -webkit-overflow-scrolling: touch;
  box-sizing: border-box;
  flex: 1;
  overflow-y: auto;
  padding: $spacer;
  position: relative;

  @media screen and (min-width: $width-mobile) {
    padding: $spacer * 2 $spacer;
  }

  @media print {
    padding: 0;
  }
}

.root.centerContent {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
