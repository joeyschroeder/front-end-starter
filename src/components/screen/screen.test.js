import React from 'react';
import { Screen } from './screen';
import { create } from 'react-test-renderer';

describe('<Screen />', () => {
  it('should render correctly', () => {
    const component = create(<Screen />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
