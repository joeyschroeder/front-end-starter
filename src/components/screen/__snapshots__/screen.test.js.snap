// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should render correctly 1`] = `
<div
  className="animated fadeIn root"
/>
`;
